lib/ansible/modules/network/ovs/openvswitch_db.py

```diff
@@ -125,8 +144,15 @@ def map_config_to_obj(module):
     match = re.search(r'^' + module.params['col'] + r'(\s+):(\s+)(.*)$', out, re.M)
 
     col_value = match.group(3)
+
+    # Map types require key argument
+    has_key = module.params['key'] is not None
+    is_map = col_value[0] == '{'
+    if is_map and not has_key:
+        module.fail_json(msg="missing required arguments: key")
+
     col_value_to_dict = {}
-    if col_value and col_value != '{}':
+    if MAP_RE.match(col_value):
```

Comment:
What is the need for this re again ? 'is_map' can be used to figure if column value is a map ?

Reply:
That's to decide whether the column is a non-empty map. is_map doesn't take into account the content of the map